pkg/lib/config/config.go

```diff
@@ -55,9 +56,10 @@ type AppConfig struct {
 	HTTP *HTTPConfig `json:"http"`
 	Hook *HookConfig `json:"hook,omitempty"`
 
-	UI           *UIConfig           `json:"ui,omitempty"`
-	Localization *LocalizationConfig `json:"localization,omitempty"`
-	Messaging    *MessagingConfig    `json:"messaging,omitempty"`
+	UI                       *UIConfig                 `json:"ui,omitempty"`
+	Localization             *LocalizationConfig       `json:"localization,omitempty"`
+	Messaging                *MessagingConfig          `json:"messaging,omitempty"`
+	CustomSMSProviderConfigs *CustomSMSProviderConfigs `json:"custom_sms_provider,omitempty"`
```

Comment:
One quick comment first, we further review the PR later 🙏🏻

We have to put the custom sms provider config to the `authgear.secrets.yaml`. The reasons:

1. To be consistent with the other providers (twilio and nexmo)
2. We have `build-in`, `custom`, and `app` resources. `custom` is the resource defined per cluster, `app` is the resource defined per app. Different resources (`authgear.yaml`, `authgear.secrets.yaml`...etc) have their own way to load the resource. For `authgear.secrets.yaml`, we support combining `custom` and `app` resources. That means we can provide a custom SMS provider config in `authgear.secrets.yaml` of the custom resource,  then the whole cluster will be able to use the custom SMS provider. This is how we config the production cluster to use Twilio for all projects.

Let me know if further explanation is needed.

Reply:
@carmenlau
I thought we are going to have a custom `authgear.yaml` defined for a certain app.
See if I understand it correctly: In production we want to share the same `custom_sms_provider` of `authgear.secrets.yaml` config, but we will configure the sms provider used (twilio / custom) per app? So I should put `custom_sms_provider` into secrets we don't have to add this config to each app?

Reply:
@tung2744 Actually we don’t have this use case now (share custom sms provider in production). Putting this to authgear.secrets.yaml is just for future flexibility. Maybe consistency is the stronger reason.

Any reason you want to put it in authgear.yaml instead?

Reply:
@carmenlau The reason is there is no secrets in the config, but I think I can still move it to `authgear.secrets.yaml`.